### PR TITLE
[PCFG-14] added pipeline configuration for `@pagopa/pagopa-api-config` repo

### PR DESCRIPTION
**Pull request by jacopocarlini**
**List of changes**
- added pipelines for `pagopa-api-config`;

### CODE REVIEW

**Review by pasqualedevita — APPROVED** [review]
thanks @jacopocarlini I make little changes to use terraform modules with more compact configuration.
I removed unused PROD connections
@jacopocarlini your PR is now applied can u check code review pipelines? When all is ok we can merge

File: azure-devops/projects/pagopa-projects/pagopa-api-config.tf [hunk 1]
```diff
@@ -1,281 +1,102 @@
...
-
-  variable {
-    name           = "GITHUB_CONNECTION"
-    value          = azuredevops_serviceendpoint_github.io-azure-devops-github-rw.service_endpoint_name
-    allow_override = false
+locals {
+  # global vars
+  pagopa-api-config-variables = {
+    cache_version_id = "v1"
+    default_branch   = var.pagopa-mock-ec.repository.branch_name
```
Comment (jacopocarlini) [inline azure-devops/projects/pagopa-projects/pagopa-api-config.tf:28]: this variable is wrong. 
The right value is: `default_branch   = var.pagopa-api-config.repository.branch_name`

**Comment by jacopocarlini** [comment]
> thanks @jacopocarlini I make little changes to use terraform modules with more compact configuration.
> I removed unused PROD connections
> @jacopocarlini your PR is now applied can u check code review pipelines? When all is ok we can merge

@pasqualedevita can you re-apply this configuration?

File: azure-devops/projects/pagopa-projects/pagopa-api-config.tf (hunk 1 above)
Comment (pasqualedevita) [reply azure-devops/projects/pagopa-projects/pagopa-api-config.tf]: sure, done

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**